pkg/scheduler/framework/session_plugins.go

```diff
@@ -801,8 +823,10 @@ func (ssn *Session) BuildVictimsPriorityQueue(victims []*api.TaskInfo) *util.Pri
 
 		lvJob, lvJobFound := ssn.Jobs[lv.Job]
 		rvJob, rvJobFound := ssn.Jobs[rv.Job]
-		if lvJobFound && rvJobFound && lvJob.Queue != rvJob.Queue {
-			return !ssn.QueueOrderFn(ssn.Queues[lvJob.Queue], ssn.Queues[rvJob.Queue])
+		preemptorJob, preemptorJobFound := ssn.Jobs[preemptor.Job]
+
+		if lvJobFound && rvJobFound && preemptorJobFound && lvJob.Queue != rvJob.Queue {
```

Comment:
What problem is this intended to solve?

Reply:
The main consideration here is the need to prioritize the two queues based on the information from the preemptor queue. Priority is given to the queue that belongs to the same parent queue as the preemptor queue.

Reply:
For the capacity plugin, the relevant `queueOrderFns` are registered both when the hierarchical queue feature is enabled and disabled. The `victimQueueOrderFn` is used here because when hierarchical queues are enabled, the sorting of victims requires information about the preemptor. Considering the need for compatibility with other plugins, in cases where no `victimQueueOrderFn` is registered, the sorting of victims ultimately uses `!QueueOrderFn`. This can be seen in the implementation of the `VictimQueueOrderFn` function.

I'm not sure if I have explained this clearly. If you have any further questions, please feel free to ask.